### DISCUSSION TITLE
Incorrect row number when extract tables

### DISCUSSION THREAD

**Discussion by tujinshu** [jsvine/pdfplumber#924] [Ask for help with specific PDFs]
## Describe the bug

*A clear and concise description of what the bug is.*
Incorrect row number in green cycle

<img width="526" alt="企业微信截图_b25fcb8e-2910-4580-b734-7eded7cfeb8c" src="https://github.com/jsvine/pdfplumber/assets/7986082/088bbf2a-f81b-4c40-a4bc-b04ef1abe0e6">

actual data （page 45）
![image](https://github.com/jsvine/pdfplumber/assets/7986082/6af211b5-7291-454b-b951-5a521a022499)

## Code to reproduce the problem

*Paste it here, or attach a Python file.*
```
import pdfplumber
pdf = pdfplumber.open("./wps.pdf")
p0 = pdf.pages[44]
im = p0.to_image()
im.debug_tablefinder()
```

## PDF file

origin pdf
[wps.pdf](https://github.com/jsvine/pdfplumber/files/11943479/wps.pdf)

## Expected behavior

*What did you expect the result __should__ have been?*
only one row should be extracted

## Actual behavior

*What actually happened, instead?*
split to 5 rows

## Screenshots

*If applicable, add screenshots to help explain your problem.*
<img width="526" alt="企业微信截图_b25fcb8e-2910-4580-b734-7eded7cfeb8c" src="https://github.com/jsvine/pdfplumber/assets/7986082/088bbf2a-f81b-4c40-a4bc-b04ef1abe0e6">

## Environment

- pdfplumber version: [0.9.0]
- Python version: [3.8.0]
- OS: [Linux]

## Additional context

*Add any other context/notes about the problem here.*

**Comment by jsvine** [comment]
Hi @tujinshu, and thanks for your interest in this library. The situation you're encountering is fairly common, and typically is caused by "invisible" graphics (rectangles, lines, etc.) on the page. In this particular example, it's a series of rectangles:

```python
im.reset().draw_rects(p0.rects)
```

![image](https://github.com/jsvine/pdfplumber/assets/534702/2950f406-e196-48ba-9633-3c8b78de0b40)

In this particular situation, these rectangles can be identified at least a couple of ways:

- Their `non_stroking_color` is `1` instead of `0`
- Their `height` and `width` are both greater than `1`. (The other `rects` are very narrow, making them look more like lines.)

To demonstrate:

```python
im.reset().draw_rects([ r for r in p0.rects if (r["width"] > 1) and (r["height"] > 1)])
```

![image](https://github.com/jsvine/pdfplumber/assets/534702/247448da-02ee-4479-9044-f3b9bc3d7c1a)

To ignore these invisible rectangles when extracting the tables, you'll want to use the `page.filter(...)` method. For examples of how this works, see the solutions to these discussions:

- https://github.com/jsvine/pdfplumber/discussions/895
- https://github.com/jsvine/pdfplumber/issues/311
